File: Lua/Pseudocode/5 Problems/AddManyNums.lua
--Take unknown number of numbers from user, add together, print result
function addManyNums()
    -- variables
    local sum = 0
    local getNextNumber = true
    -- Get the first number
    while getNextNumber == true do
        -- variables
        local inputNumber = nil

        -- Get number to add to sum
        while (inputNumber == nil) do --Keep asking user until input is valid
            -- Use tonumber to handle invalid input on store
            -- Invalid numbers return nil
            print("Number: ")
            inputNumber = tonumber(io.read())
        end
        -- Add number to sum
        sum = sum + inputNumber

        -- Ask to add another number
        print("Add another number? (y/n) ")
        if (string.upper(io.read()) ~= "Y") then
            getNextNumber = false
        end
    end
    -- Output sum
    print("Output: "..sum)
end

File: Python/Pseudocode/5 Problems/AddManyNums.py
# Get an unknown amount of numbers from the user and add them together, print out the result.

# Is the input a number
def IsNum(userInput):
    try:
        float(userInput)
        isFloat = True
    except:
        isFloat = False

    return isFloat

# While not a number, keep promting
def TryAgain(userInput):
    while not IsNum(userInput):
        userInput = input("That is not a number, please try again: ")
    return userInput

# Ask user if they want to go again
def CheckContinue():
    userInput = ""
    # Ask to add another number
    while userInput != "Y" and userInput != "N":
        # Store uppercase for all checks
        userInput = str.upper(input("Add another number? (y/n): "))
    # Ouput true if Y was entered
    return userInput == "Y"
        
# Get user input and return sum with user input
def AddNumber(sum):
    # Get Number
    userInput = input("Enter a number to add: ")
    userInput = TryAgain(userInput)
    # Add to sum
    sum += float(userInput)
    return sum


# Initialize variables
totalSum = AddNumber(0)
# Get and add numbers
while CheckContinue():
    totalSum = AddNumber(totalSum)

print("Output: " + str(totalSum))
